### PR TITLE
Marcelbrucker code rework

### PR DESCRIPTION
**Pull request by marcelbrucker**
- Carried out
> general: param_server["Scenario"]["Generation"]["InteractionDatasetScenarioGeneration"][....] can be shortened using local_params = param_server["Scenario"]["Generation"]["InteractionDatasetScenarioGeneration"] plus local_params[...]
-  correct some spelling mistakes in the notebook
- added explanations for states and actions in notebook

### CODE REVIEW

File: bark_ml/library_wrappers/lib_tf2rl/agents/gail_agent.py [hunk 1]
```diff
@@ -22,9 +22,15 @@ def __init__(self,
 
         Args:
             environment (Runtime, optional): A environment with a gym
+<<<<<<< HEAD
```
Comment (pedroacvieira) [inline bark_ml/library_wrappers/lib_tf2rl/agents/gail_agent.py]: You forgot a conflict here. It needs solving :P

File: bark_ml/library_wrappers/lib_tf2rl/agents/gail_agent.py [hunk 2]
```diff
@@ -38,37 +44,46 @@ def _get_generator(self):
...
         """
+        local_params = self._params["ML"]["BehaviorGAILAgent"]
+
         irl = GAIL(
             state_shape=self._environment.observation_space.shape,
             action_dim=self._environment.action_space.high.size,
-            units=self._params["ML"]["BehaviorGAILAgent"]["Discriminator"]["FcLayerParams", "", [
+            units=local_params["Discriminator"]["FcLayerParams", "", [
                 400, 300]],
+<<<<<<< HEAD
```
Comment (pedroacvieira) [inline bark_ml/library_wrappers/lib_tf2rl/agents/gail_agent.py]: Another conflict here

**Review by pedroacvieira — CHANGES_REQUESTED** [review]
Except for two forgotten conflicts, everything is alright :D

File: bark_ml/library_wrappers/lib_tf2rl/agents/gail_agent.py [hunk 3]
```diff
@@ -38,37 +38,39 @@ def _get_generator(self):
         """Returns instantiated policy -
         parameters from ./examples/example_params/gail_params.json
         """
+        local_params = self._params["ML"]["BehaviorGAILAgent"]
```
Comment (marcelbrucker) [inline bark_ml/library_wrappers/lib_tf2rl/agents/gail_agent.py]: Suggestion of Tobias.

File: bark_ml/library_wrappers/lib_tf2rl/runners/gail_runner.py [hunk 4]
```diff
@@ -102,33 +102,35 @@ def _get_args_from_params(self):
         - gpu:                    int, name of gpu device
 
     """
+    local_params = self._params['ML']['GAILRunner']
```
Comment (marcelbrucker) [inline bark_ml/library_wrappers/lib_tf2rl/runners/gail_runner.py]: Suggestion of Tobias.

File: docs/report/README.md [hunk 5]
```diff
@@ -1,7 +1,5 @@
 # Run the Jupyter Server
 
-First extract the files in `examples/expert_trajectories.zip` to `docs/report/data/expert_trajectories`. Then start up the jupyter notebook with:
```
Comment (marcelbrucker) [inline docs/report/README.md:3]: Not needed anymore.

File: bark_ml/library_wrappers/lib_tf2rl/agents/gail_agent.py (hunk 3 above)
Comment (MarcelBruckner) [reply bark_ml/library_wrappers/lib_tf2rl/agents/gail_agent.py]: Can you please make it even one deeper. So down to `self._params["ML"]["BehaviorGAILAgent"]["Generator"]` and rename to `generator_params`? And then clearly adjust the `local_params["Generator"]` to be only `generator_params`

File: bark_ml/library_wrappers/lib_tf2rl/agents/gail_agent.py [hunk 6]
```diff
@@ -38,37 +38,39 @@ def _get_generator(self):
...
+            critic_units=local_params["Generator"]["CriticJointFcLayerParams", "", [
                 400, 300]],
-            sigma=self._params["ML"]["BehaviorGAILAgent"]["Generator"]["Sigma", "", 0.1],
-            tau=self._params["ML"]["BehaviorGAILAgent"]["Generator"]["Tau", "", 0.005],
-            n_warmup=self._params["ML"]["BehaviorGAILAgent"]["WarmUp", "", 1000],
-            batch_size=self._params["ML"]["BehaviorGAILAgent"]["Generator"]\
-                ["BatchSize", "", 100],
+            sigma=local_params["Generator"]["Sigma", "", 0.1],
+            tau=local_params["Generator"]["Tau", "", 0.005],
+            n_warmup=local_params["WarmUp", "", 1000],
```
Comment (MarcelBruckner) [inline bark_ml/library_wrappers/lib_tf2rl/agents/gail_agent.py]: Can we move the `WarmUp` to the `["Generator"]` part in the params file, as it is only used here once?

File: bark_ml/library_wrappers/lib_tf2rl/agents/gail_agent.py [hunk 7]
```diff
@@ -38,37 +38,39 @@ def _get_generator(self):
...
+            batch_size=local_params["Generator"]["BatchSize", "", 100],
             gpu=self._params["ML"]["Settings"]["GPUUse", "", 0])
         return policy
 
     def _get_discriminator(self):
         """Returns instantiated discriminator network -
         parameters from ./examples/example_params/gail_params.json
         """
+        local_params = self._params["ML"]["BehaviorGAILAgent"]
```
Comment (MarcelBruckner) [inline bark_ml/library_wrappers/lib_tf2rl/agents/gail_agent.py]: Same as above but with the `["Discriminator"]`

File: bark_ml/library_wrappers/lib_tf2rl/agents/gail_agent.py [hunk 8]
```diff
@@ -38,37 +38,39 @@ def _get_generator(self):
...
             action_dim=self._environment.action_space.high.size,
-            units=self._params["ML"]["BehaviorGAILAgent"]["Discriminator"]["FcLayerParams", "", [
+            units=local_params["Discriminator"]["FcLayerParams", "", [
                 400, 300]],
-            lr=self._params["ML"]["BehaviorGAILAgent"]["Discriminator"]["LearningRate", "", 0.001],
-            enable_sn=self._params["ML"]["BehaviorGAILAgent"]["EnableSN", "", False],
-            batch_size=self._params["ML"]["BehaviorGAILAgent"]["Discriminator"]\
-                ["BatchSize", "", 32],
+            lr=local_params["Discriminator"]["LearningRate", "", 0.001],
+            enable_sn=local_params["EnableSN", "", False],
```
Comment (MarcelBruckner) [inline bark_ml/library_wrappers/lib_tf2rl/agents/gail_agent.py]: Also move as above into the `["Discriminator"]`

File: bark_ml/library_wrappers/lib_tf2rl/runners/gail_runner.py (hunk 4 above)
Comment (MarcelBruckner) [reply bark_ml/library_wrappers/lib_tf2rl/runners/gail_runner.py]: Same as above but with `["tf2rl"]`

**Review by MarcelBruckner — CHANGES_REQUESTED** [review]
There are some parts that can be made even better accessible. I commented the parts I think should be changed.

**Comment by MarcelBruckner** [comment]
I just saw that there are many more locations where we can apply this shortening:
- bark_ml/library_wrappers/lib_tf2rl/generate_expert_trajectories.py
- bark_ml/tests/py_library_tf2rl_tests/base_tests.py
- bark_ml/tests/py_library_tf2rl_tests/simulation_based_tests.py
- bark-ml/docs/report/gail.ipynb
- Some more

You can find them with `ctrl + shift + f` to global search for `param_server`

**Comment by marcelbrucker** [comment]
> I just saw that there are many more locations where we can apply this shortening:
> 
> * bark_ml/library_wrappers/lib_tf2rl/generate_expert_trajectories.py
> * bark_ml/tests/py_library_tf2rl_tests/base_tests.py
> * bark_ml/tests/py_library_tf2rl_tests/simulation_based_tests.py
> * bark-ml/docs/report/gail.ipynb
> * Some more
> 
> You can find them with `ctrl + shift + f` to global search for `param_server`

I don't see how this should work in most of your suggested files when a param_server is filled with values (<--) instead of accessing many values from the same block of a param_server (-->).
But I did it once more in * bark_ml/library_wrappers/lib_tf2rl/generate_expert_trajectories.py .

**Comment by MarcelBruckner** [comment]
> > I just saw that there are many more locations where we can apply this shortening:
> > 
> > * bark_ml/library_wrappers/lib_tf2rl/generate_expert_trajectories.py
> > * bark_ml/tests/py_library_tf2rl_tests/base_tests.py
> > * bark_ml/tests/py_library_tf2rl_tests/simulation_based_tests.py
> > * bark-ml/docs/report/gail.ipynb
> > * Some more
> > 
> > You can find them with `ctrl + shift + f` to global search for `param_server`
> 
> I don't see how this should work in most of your suggested files when a param_server is filled with values (<--) instead of accessing many values from the same block of a param_server (-->).
> But I did it once more in * bark_ml/library_wrappers/lib_tf2rl/generate_expert_trajectories.py .

Like this: https://github.com/GAIL-4-BARK/bark-ml/pull/68/commits/2afc0c1eadebe9739dfdb5ffa0b786104ef313ef
I already changed it so that I can now merge it :+1:

**Pull request merged**